Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/measure.h

```diff
@@ -334,29 +338,22 @@ longest_border(const PolygonMesh& pmesh,
             typename property_map_value<PolygonMesh, CGAL::vertex_point_t>::type>::Kernel::FT  FT;
   typedef typename boost::graph_traits<PolygonMesh>::halfedge_descriptor                       halfedge_descriptor;
 
-  std::unordered_set<halfedge_descriptor> visited;
+  std::deque<halfedge_descriptor> boundary_cycles;
```

Comment:
why use a deque?

Reply:
to avoid copying when reallocations happen.

Reply:
Wouldn't this require an absurd number of boundary CCs for this to start mattering?

Reply:
you are right, but on the other hand what is wrong about a deque.

Reply:
When I see a deque rather than a vector, I expect the deque's specific toolbox to come into play, and I would wonder if I missed something if it is not used. 

It's not very important.